### PR TITLE
setup.py: set an upper bound on dependencies version

### PR DESCRIPTION
**Pull request by JordanP**
This should make this library more robust in case of breaking changes in one of
its dependencies.

### CODE REVIEW

**Comment by moggers87** [comment]
LXML 4.0 was released in 2017. I assume that was a typo and you meant to put 5.0?

**Comment by JordanP** [comment]
@moggers87 yeah, I think it was a typo. 

Let's first agree or not, whether this project should set a some major version boundaries on its dependencies. What do you think ?

**Comment by moggers87** [comment]
I'm generally against the idea, unless a particular project is known to introduce breaking changes on major versions (e.g. Django, Celery). While users might get caught out by some unexpected changes, that is something the user can work around until there's a new release of premailer (i.e. they can set `lxml<5.0` in their requirements file).

If we do limit dependencies in this way we might end up locking users into old versions of libraries that have missing features, bugs, and security issues. Users would then have to wait for a new premailer release.

tl;dr: unless we know better, I think it's best to leave pinning dependencies to users

**Comment by JordanP** [comment]
I understand. 

More or more libraries are following semver where a major version bumps means a non backward compatible. API. Agreed that, users can adjust their dependencies and say set `lxml<5.0 in their requirements file` but in a average Python project you get dozens of dependencies so easily a hundred of indirect dependencies, not convenient. 

But it's a tradeoff and your stand is totally valid and shared by many too. .
